src/meta/src/barrier/mod.rs

```diff
@@ -234,7 +234,7 @@ impl CheckpointControl {
     async fn finish_commands(&mut self, checkpoint: bool) -> MetaResult<bool> {
         for command in self
             .finished_commands
-            .extract_if(|c| checkpoint || c.is_barrier())
+            .extract_if(|c| checkpoint || !c.is_checkpoint())
```

Comment:
We might consider renaming these variables with more descriptive names to avoid confusion and improve readability. Every time I encounter these variables, I spend considerable time ensuring that I have understood them correctly.

Reply:
Good suggestion. Made the change.